doc/specifications/rest/REST-API-V2.rst

```diff
@@ -4888,7 +4890,28 @@ List Users
           User_
 
 :Error Codes:
-    :401: If the user has no permission to read users
+    :404: If there are no visibile users matching the filter
```

Comment:
I'm a bit unsure about this change, for two reasons:
- 401 as I read it is a bout lack of permissions to read users at all, we could for instance check if user has rights to read any users at all early in the request, and if users are ever moved out of content tree it will have it's own module/policy making this more inline
- I don't think the 401 was intended to be thrown if no users where found, o I think 401 should still be in the spec, however given the nature of this endpoint I'm struggling a bit with agreeing with myself if no users should be 404 or just a empty list of user refs. As a developer I would expect 404 on unique id lookups, but empty lists on group lookups like roleId. However maybe I'm thinking to much PHP api here, and also given both cases are on the same endpoint I of course prefer it is consistent.


So to sum up:
- Keep 401 imho, as it from spec perspective seems to be about permissions, not NotFound cases. can be unimplemented atm.
- Add 404 as you did, or return empty list, I think you and @dpobel knows the needs of the ui better here and what works best in the REST api.